calyx/src/default_passes.rs

```diff
@@ -74,10 +75,11 @@ impl PassManager {
                 RemoveCombGroups, // Must run before `infer-static-timing`.
                 InferStaticTiming,
                 MergeStaticPar,
-                DeadGroupRemoval,
-                StaticParConv, // Must be before `collapse-control`
+                DeadGroupRemoval, // Sincce MergeStaticPar potentialy creates dead groups
+                StaticParConv,    // Must be before `collapse-control`
                 CollapseControl,
                 CompileRef, //Must run before 'resource-sharing'.
+                GroupToSeq,
                 CellShare,
```

Comment:
Open an issue saying that we should run CellShare as early as possible, ideally before `merge-static-par`. This cannot be done currently because it doesn't handle `ref cells` passed into an invoke: https://docs.calyxir.org/lang/memories-by-reference.html#the-easy-way